panic: runtime error: invalid memory address or nil pointer dereference

<!-- Please use this template for reporting bugs and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!-->



When I pass   sudo devspace create space my-app, at the end, launch the message: 

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xb0 pc=0x16b4d2d]

goroutine 1 [running]:
github.com/devspace-cloud/devspace/pkg/devspace/cloud.(*provider).Login(0xc00016bd40, 0x0, 0x0)
	/Users/runner/runners/2.162.0/work/devspace/devspace/pkg/devspace/cloud/login.go:91 +0x39d
github.com/devspace-cloud/devspace/pkg/devspace/cloud.GetProviderWithOptions(0x0, 0x0, 0x0, 0x0, 0x40c800, 0x2956cc0, 0xc0017a0060, 0x29bf1c0, 0x4054120, 0xc0017b0180, ...)
	/Users/runner/runners/2.162.0/work/devspace/devspace/pkg/devspace/cloud/provider.go:123 +0xa65
github.com/devspace-cloud/devspace/pkg/devspace/cloud.GetProvider(0x0, 0x0, 0x29bf1c0, 0x4054120, 0x1, 0xc0000ceea0, 0xc0012fbc08, 0xe70c9b)
	/Users/runner/runners/2.162.0/work/devspace/devspace/pkg/devspace/cloud/provider.go:52 +0x89
github.com/devspace-cloud/devspace/cmd/create.(*spaceCmd).RunCreateSpace(0xc000ef4390, 0xc000efe000, 0xc0002e9010, 0x1, 0x1, 0x0, 0x0)
	/Users/runner/runners/2.162.0/work/devspace/devspace/cmd/create/space.go:62 +0x1b6
github.com/spf13/cobra.(*Command).execute(0xc000efe000, 0xc0002e8fb0, 0x1, 0x1, 0xc000efe000, 0xc0002e8fb0)
	/Users/runner/runners/2.162.0/work/devspace/devspace/vendor/github.com/spf13/cobra/command.go:826 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0xc000ef2000, 0x40aea30, 0xc000ef2000, 0x0)
	/Users/runner/runners/2.162.0/work/devspace/devspace/vendor/github.com/spf13/cobra/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/runner/runners/2.162.0/work/devspace/devspace/vendor/github.com/spf13/cobra/command.go:864
github.com/devspace-cloud/devspace/cmd.Execute()
	/Users/runner/runners/2.162.0/work/devspace/devspace/cmd/root.go:70 +0xbd
main.main()
	/Users/runner/runners/2.162.0/work/devspace/devspace/main.go:17 +0x3e



- Operating System: linux 




<!-- DO NOT EDIT BELOW THIS LINE -->
/kind bug

--- a/pkg/devspace/cloud/login.go
+++ b/pkg/devspace/cloud/login.go
@@ -88,6 +88,10 @@ func (p *provider) Login() error {
 
 		// Check if we got access
 		p.Key = key
+		if p.client == nil {
+			p.client = client.NewClient(p.Name, p.Host, key, "")
+		}
+
 		_, err := p.client.GetSpaces()
 		if err != nil {
 			close(keyChannel)

--- a/pkg/devspace/cloud/provider.go
+++ b/pkg/devspace/cloud/provider.go
@@ -112,7 +112,7 @@ func GetProviderWithOptions(useProviderName, key string, relogin bool, loader co
 
 		if key != "" {
 			provider.Key = key
-			provider.client = client.NewClient(providerName, p.Host, key, p.Token)
+			provider.client = client.NewClient(providerName, p.Host, key, "")
 
 			// Check if we got access
 			_, err := provider.client.GetSpaces()

